refactor(alert): remove intermediate 'div' element

maxokorokov opened the pull request:
Fixes #2544

pkozlowski-opensource commented on src/alert/alert.ts:
```diff
@@ -38,12 +43,22 @@ export class NgbAlert {
+  ngOnChanges(changes: SimpleChanges) {
+    const {type} = changes;
+    if (type && !type.firstChange) {
+      this._renderer.removeClass(this._element.nativeElement, `alert-${type.previousValue}`);
```
Why do we do this through low-level DOM manipulation? Couldn't we move the existing bindings to the host?
Or is binding overriding classes set by a user on a host?

maxokorokov replied:
Double-checked, and it overrides user-set classes. So I see three options:

1. Leave as it is done currently with `Renderer2`
2. hardcode all supported classes and remove the demo with `alert-customclass`:

```
  host: {
    'role': 'alert',
    'class': 'alert d-block',
    '[class.alert-dismissible]': 'dismissible',
    '[class.alert-success]': `type === 'success'`,
    '[class.alert-warning]': `type === 'warning'`,
    '[class.alert-info]': `type === 'info'`,
    '[class.alert-danger]': `type === 'danger'`,
    '[class.alert-primary]': `type === 'primary'`,
    '[class.alert-secondary]': `type === 'secondary'`,
    '[class.alert-light]': `type === 'light'`,
    '[class.alert-dark]': `type === 'dark'`
  },
```
3. do nothing and close this :)

pkozlowski-opensource commented on src/alert/alert.ts:
```diff
@@ -38,12 +43,22 @@ export class NgbAlert {
   closeHandler() { this.close.emit(null); }
+
+  ngOnChanges(changes: SimpleChanges) {
+    const {type} = changes;
```
While it works I find the `type` name confusing / not very descriptive  here. A much better name would be `typeChange`. I would prefer to be explicit here.

maxokorokov replied:
Will change

pkozlowski-opensource commented on src/alert/alert.ts:
```diff
@@ -38,12 +43,22 @@ export class NgbAlert {
+    }
+  }
+
+  ngOnInit() { this._renderer.addClass(this._element.nativeElement, `alert-${this.type}`); }
```
Couldn't we move this logic to `ngOnChanges` and set the class on `firstChange`? This way the whole logic would be centralised in one place

pkozlowski-opensource commented:
@maxokorokov I left few nits but generally speaking I think that we should merge it. It is not an ideal solution but this is the best we've got given current Angular...

maxokorokov commented:
@pkozlowski-opensource , fixed comments and:
- added a `BREAKING CHANGE` comment about the markup change
- replaced `d-block` with `:host { display: block; }` as it doesn't set `!important` flag
